### PR TITLE
ToC shows where you are as you scroll

### PR DESCRIPTION
**Pull request by molebox**
The table of contents now shows where you are on the page. 

toc-animation

The problem is that the content in the sidebar is too long and we have issues with the stickness of the sidebar

### CODE REVIEW

File: src/components/toc.tsx [hunk 1]
```diff
@@ -41,30 +41,87 @@ const TOCContainer = styled.div`
   top: 10px;
 `
 
+interface ItemProps {
+  isActive: boolean
+}
+
+const ListItem = styled.a<ItemProps>`
+  border-bottom: ${props => (props.isActive ? '1px solid' : '')};
```
Comment (hrescak) [inline src/components/toc.tsx]: I'd do this to highlight the current item:
``` 
const ListItem = styled.a<ItemProps>`
  transition: all 0.2s ease-out;
  ${props => props.isActive ? 'text-color: theme.colors.gray900;' : ''}
`
```

Comment (molebox) [reply src/components/toc.tsx]: Ohhh smart, yeah i couldnt figure out how to animate the border without a hover action. Im not sure how well that plays with the `inlinecode`. Will check and see.

Comment (hrescak) [reply src/components/toc.tsx]: Yeah I totally just did a github yolo coding so I only have a vague idea how this is going to look 😆

Comment (molebox) [reply src/components/toc.tsx]: > g to look 😆

Best kind of coding 👯‍♀️

**Comment by molebox** [comment]
This depends on #1553 getting merged

**Comment by molebox** [comment]
This looks great @nilubava! I'm so happy you managed to solve the scrolling issue! This PR was opened by me and as such I cannot approve my own PRs 😅 But I would approve this if I could 👍

**Review by nilubava — APPROVED** [review]

**Pull request merged**